dev/ScrollPresenter/APITests/ScrollPresenterTests.cs

```diff
@@ -879,6 +879,40 @@ public void ListenToContentEffectiveViewportChanged()
             });
         }
 
+        [TestMethod]
+        [TestProperty("Description", "ScrollPresenter should handle BringIntoView for its direct content.")]
+        public void HandleContentBringIntoView()
+        {
+            ScrollPresenter scrollPresenter = null;
+            Rectangle rectangleScrollPresenterContent = null;
+            AutoResetEvent scrollPresenterViewChangedEvent = new AutoResetEvent(false);
+            AutoResetEvent scrollPresenterLoadedEvent = new AutoResetEvent(false);
+
+            RunOnUIThread.Execute(() =>
+            {
+                rectangleScrollPresenterContent = new Rectangle() { Margin = new Thickness(0, 500, 0, 0) };
+                scrollPresenter = new ScrollPresenter();
+
+                SetupDefaultUI(scrollPresenter, rectangleScrollPresenterContent, scrollPresenterLoadedEvent);
+
+            });
+
+            WaitForEvent("Waiting for Loaded event", scrollPresenterLoadedEvent);
+            
+            RunOnUIThread.Execute(() =>
+            {
+                scrollPresenter.ViewChanged += (s, e) =>
+                {
+                    scrollPresenterViewChangedEvent.Set();
+                };
+
+                rectangleScrollPresenterContent.StartBringIntoView(new BringIntoViewOptions() { AnimationDesired = false });
+
+                WaitForEvent("Waiting for ViewChanged event", scrollPresenterViewChangedEvent);
```

Comment:
Reason 1 for test failure:
WaitForEvent cannot be used inside a RunOnUIThread.Execute...
It must run off the UI thread.

Reply:
Fixed